app/services/file_generator.rb

```diff
@@ -0,0 +1,98 @@
+# typed: true
+# frozen_string_literal: true
+
+# This generates a File for a work
+class FileGenerator
+  extend T::Sig
+
+  sig do
+    params(
+      work: Work,
+      attached_file: AttachedFile
+    ).returns(T.any(Cocina::Models::File, Cocina::Models::RequestFile))
+  end
+  def self.generate(work:, attached_file:)
+    new(work: work, attached_file: attached_file).generate
+  end
+
+  sig { params(work: Work, attached_file: AttachedFile).void }
+  def initialize(work:, attached_file:)
+    @work = work
+    @attached_file = attached_file
+  end
+
+  attr_reader :work, :attached_file
+
+  sig { returns(T.nilable(T.any(Cocina::Models::File, Cocina::Models::RequestFile))) }
+  def generate
+    return nil unless blob
+
+    if work.druid
+      Cocina::Models::File.new(file_attributes)
+    else
+      Cocina::Models::RequestFile.new(request_file_attributes)
+    end
+  end
+
+  def request_file_attributes
+    {
+      type: 'http://cocina.sul.stanford.edu/models/file.jsonld',
+      version: work.version,
+      label: attached_file.label,
+      filename: filename,
+      access: access,
+      administrative: administrative,
+      hasMimeType: blob.content_type,
+      hasMessageDigests: message_digests,
+      size: blob.byte_size
+    }
+  end
+
+  def file_attributes
+    request_file_attributes.merge(externalIdentifier: external_identifier)
+  end
+
+  def filename
+    blob.filename.to_s # File.basename(filename(blob.key))
```

Comment:
What's the comment here for?

Reply:
I don't know. It was in the RequestGenerator prior to this change. I assume it means that those are equivalent?

Reply:
After spending ten minutes reviewing the changes in this file, I now see that it's a refactoring, so take all review up until now with a grain of salt.